pennylane/ops/qubit/hamiltonian.py

```diff
@@ -206,8 +205,6 @@ def __init__(
                 )
 
         coeffs_flat = [self._coeffs[i] for i in range(qml.math.shape(self._coeffs)[0])]
-        # overwrite this attribute, now that we have the correct info
-        self.num_params = qml.math.shape(self._coeffs)[0]
```

Comment:
This is where the new approach shines: this hotfix of setting `num_params = 1` and then overwriting it is not not necessary

Reply:
@mariaschuld do you need to set this manually? I thought when you call `super().__init__(*coeffs)`, the logic inside of the parent `__init__` automatically sets `self._num_params`?

Reply:
Exactly, this is why this subclass does not overwrite `def num_params`! It uses the default one.

Reply:
wait, I must admit I am still confused a bit. Can't we delete this line, since on line 215, when `super().__init__` is called, this will trigger

https://github.com/PennyLaneAI/pennylane/blob/1c1cb9b67f34c7dd2e2a1b777aa597a29be9a079/pennylane/operation.py#L480

in the base class?